.cursor/rules/project-setup.mdc

```diff
@@ -0,0 +1,205 @@
+---
+description: 
+globs: 
+alwaysApply: true
+---
+---
+description: Setup and requirements for WAVS projects
+globs: **/*
+---
+# Project Structure Guidelines
+
+Guidelines for organizing WAVS projects and understanding the repository layout.
+
+## Directory Structure
+
+```
+project-root/
+├── .cursor/
+│   └── rules/              # Project-specific coding rules
+├── .docker/                # Docker configuration and deployment files
+│   ├── service.json       # Service configuration
+│   ├── deployments.json   # Contract deployment addresses
+│   └── script_deploy.json # Script deployment data
+├── components/            # WASI component implementations
+│   └── eth-price-oracle/  # Example component
+│       ├── Cargo.toml    # Rust dependencies
+│       └── src/
+│           ├── lib.rs     # Component implementation
+│           └── bindings/  # Generated WASI bindings
+├── docs/                  # Documentation files
+│   ├── tutorial/         # Step-by-step guides
+│   ├── custom-components.mdx  # Custom component documentation
+│   ├── benefits.mdx      # WAVS benefits documentation
+│   ├── design.mdx        # System design documentation
+│   ├── how-it-works.mdx  # Technical implementation details
+│   ├── index.mdx         # Documentation landing page
+│   └── overview.mdx      # WAVS platform overview
+├── src/
+│   ├── contracts/        # Solidity smart contracts
+│   │   ├── WavsTrigger.sol   # Trigger contract
+│   │   └── WavsSubmit.sol    # Submission contract
+│   └── interfaces/       # Contract interfaces
+│       └── ITypes.sol    # Shared type definitions
+├── script/              # Deployment and test scripts
+│   ├── build_components.sh  # WASI build script
+│   └── Trigger.s.sol       # Trigger deployment script
+├── compiled/            # Compiled WASM components
+├── Cargo.toml          # Workspace-level Rust dependencies
+├── Makefile           # Build and deployment commands
+└── wavs.toml         # WAVS service configuration
+```
+
+## Key Files
+
+1. Configuration Files:
+```toml
+# wavs.toml - Service configuration
+[service]
+name = "example-service"
+version = "0.1.0"
+
+[chains]
+ethereum = { rpc_url = "http://localhost:8545" }
+
+[components]
+price_oracle = { wasm = "eth_price_oracle.wasm" }
+```
```

Comment:
Think we could use a bit more info here on the `wavs.toml` file. What is it? What is it used for?